syncd/syncd.cpp

```diff
@@ -95,36 +90,20 @@ sai_object_id_t gSwitchId;
 std::string fdbFlushSha;
 std::string fdbFlushLuaScriptName = "fdb_flush.lua";
 
-struct cmdOptions
-{
-    bool diagShell;
-    bool useTempView;
-    int startType;
-    bool disableExitSleep;
-    bool enableUnittests;
-    std::string profileMapFile;
-#ifdef SAITHRIFT
-    bool run_rpc_server;
-    std::string portMapFile;
-#endif // SAITHRIFT
-    bool syncMode;
-    ~cmdOptions() {}
-};
-
-cmdOptions options;
+std::shared_ptr<CommandLineOptions> g_commandLineOptions; // TODO move to syncd object
 
 bool enableUnittests()
```

Comment:
>enableUnittests [](start = 5, length = 15)

How about remove this function, and add a member function to g_commandLineOptions?

Reply:
not followin? example ?

Reply:
I mean `g_commandLineOptions->m_enableUnittests` or `g_commandLineOptions->getEnableUnittests()`. There is no benefit to wrap in a standalone function.

---
In reply to: [352920271](https://github.com/Azure/sonic-sairedis/pull/541#discussion_r352920271) [](ancestors = 352920271)

Reply:
ahh yes! will fix

Reply:
i also later will add proper functions for each flag like getEnableUnittests(), for now it's not on main task

Reply:
done

Reply:
Function not used now, let's remove?

---
In reply to: [352932040](https://github.com/Azure/sonic-sairedis/pull/541#discussion_r352932040) [](ancestors = 352932040,352920271)

Reply:
forgot !, done